SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Build/Build.cs

```diff
@@ -89,17 +89,36 @@ public static void Main(string[] args)
             // Locate the Unreal Engine.
             Console.WriteLine("Finding Unreal Engine build.");
             string uproject = File.ReadAllText(projectFile, Encoding.UTF8);
-            string engineAssociationRegex = "..{8}-.{4}-.{4}-.{4}-.{12}.";
-            Match engineAssociationMatch = Regex.Match(uproject, engineAssociationRegex);
+
+            dynamic projectJson = JObject.Parse(uproject);
+            string engineAssociation = projectJson.EngineAssociation;
+
+            Console.WriteLine("Engine Association: " + engineAssociation);
 
             string unrealEngine = "";
 
-            // If we found an engine association within the .uproject, query the registry for the engine directory.
-            if (engineAssociationMatch.Success)
+            // If the engine association is empty then climb the parent directories of the project looking for the Unreal Engine root directory.
+            if (string.IsNullOrEmpty(engineAssociation))
             {
-                string engineAssociation = engineAssociationMatch.Value;
-                Console.WriteLine("Engine Association: " + engineAssociation);
-
+                DirectoryInfo currentDir = new DirectoryInfo(Directory.GetCurrentDirectory());
+                while (currentDir.Parent != null && string.IsNullOrEmpty(unrealEngine))
```

Comment:
The `IsNullOrEmpty` test is redundant given the `break` below

Reply:
Good point